### PR TITLE
Oe/86bz12jgz/signup step2

### PR DESCRIPTION
**Pull request by guardian-001**
## What does this do?

- Created step two of the signup stepper
- Optimized functionalities in shared component
- Created new Phone input shared component that will make the functionality easy for other developers
- Used react-native-international-phone-number to make phone number picker accept any phone number from all over the world

## Why did you do this?

- To implement the second step of the signup process as part of the application's user flow
- To improve code reusability and maintainability by optimizing shared components
- To provide a standardized and easy-to-use phone number input component for other developers
- To ensure the phone number input is versatile and can handle international numbers, improving user experience

## Who/what does this impact?

- This change impacts the signup process and any developers working on features related to user registration
- The shared phone input component may be used across various parts of the application, affecting multiple modules 

## How did you test this?

Manual testing of the signup stepper to ensure it functions correctly

image

image

### CODE REVIEW

File: src/modules/architect/signup/choose-speciality.tsx [hunk 1]
```diff
@@ -2,16 +2,20 @@ import React, { useState } from 'react';
 
 import { HouseModel, InteriorHouseModel } from '@/assets/icons/archimatch';
 import { translate } from '@/core';
-import StepperButton from '@/modules/shared/stepper-button';
+import { StepButtons } from '@/modules/shared'; // Adjust the import path if necessary
```
Comment (archimatchproject) [inline src/modules/architect/signup/choose-speciality.tsx:5]: remove gpt comment

File: src/modules/architect/signup/create-profile.tsx [hunk 2]
```diff
@@ -1,16 +1,35 @@
...
+  handleNextStep?: () => void;
 };
 
-export default function CreateProfile({ onSubmit }: ResetFormProps) {
+export default function CreateProfile({
+  handlePreviousStep,
+  handleNextStep,
+}: ResetFormProps) {
+  const { control } = useCustomForm(SignupFormSchema);
+  // const router = useRouter();
```
Comment (archimatchproject) [inline src/modules/architect/signup/create-profile.tsx:23]: remove useless comments

File: src/modules/architect/signup/signup-stepper.tsx [hunk 3]
```diff
@@ -19,24 +14,54 @@ type Props = {};
 export default function SignupStepper({}: Props) {
   const space = useRouteName();
   const [step, setStep] = React.useState(0);
-
+  const route = useRouter();
   const handleNextStep = () => {
```
Comment (archimatchproject) [inline src/modules/architect/signup/signup-stepper.tsx:18]: u can extract stepper code into a seperate custom hook and use it in other use cases not only this one

File: src/modules/architect/signup/signup-stepper.tsx [hunk 4]
```diff
@@ -49,8 +74,8 @@ export default function SignupStepper({}: Props) {
 
   const { component } = stepsContent[step];
   return (
-    <KeyboardAvoidingView
-      behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
+    <View
+      // behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
```
Comment (archimatchproject) [inline src/modules/architect/signup/signup-stepper.tsx:78]: if ur not using it just delete it

File: src/modules/login/login-form.tsx [hunk 5]
```diff
@@ -23,17 +23,14 @@ export const LoginForm = ({ onSubmit }: LoginFormProps) => {
...
-  const space = useRouteName();
-
   const handleFormSubmit: SubmitHandler<LoginFormType> = (data) => {
     signIn({ access: 'access-token', refresh: 'refresh-token' });
     router.push(`/(${space})/(private)`);
     onSubmit(data);
-
-  }; 
+  };
   const handleResetPass = () => {
```
Comment (archimatchproject) [inline src/modules/login/login-form.tsx:31]: change the name to handleResetPassword

File: src/shared/components/controlled-phone-number-input.tsx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+          onChangePhoneNumber={field.onChange}
+          selectedCountry={selectedCountry}
+          onChangeSelectedCountry={handleSelectedCountry}
+          phoneInputStyles={{
+            container: { height: 35, backgroundColor: colors.white },
+            input: { fontSize: 12 },
+            callingCode: { fontSize: 12 },
+            caret: { fontSize: 12, backgroundColor: colors.white },
+            flag: { fontSize: 12 },
+            flagContainer: { width: '40%', backgroundColor: colors.white },
```
Comment (archimatchproject) [inline src/shared/components/controlled-phone-number-input.tsx:59]: double check width for tablet screen here is 40% the correct number

File: src/validations/shared/shared-schema.tsx [hunk 7]
```diff
@@ -36,7 +38,12 @@ export const LoginFormSchema = z.object({
 });
 
 export const SignupFormSchema = z.object({
+  name: fieldValidation,
+  surname: fieldValidation,
   email: emailValidation,
+  phone: phoneValidation,
+  address: fieldValidation,
+  matricule: fieldValidation, //matriculeValidation
```
Comment (archimatchproject) [inline src/validations/shared/shared-schema.tsx:46]: same thing here

**Review by archimatchproject — CHANGES_REQUESTED** [review]
left some comments

File: src/modules/architect/signup/choose-speciality.tsx (hunk 1 above)
Comment (guardian-001) [reply src/modules/architect/signup/choose-speciality.tsx:5]: khalil is advice to use it hhhh, removed

File: src/modules/architect/signup/create-profile.tsx (hunk 2 above)
Comment (guardian-001) [reply src/modules/architect/signup/create-profile.tsx:23]: fixed

File: src/modules/architect/signup/signup-stepper.tsx (hunk 4 above)
Comment (guardian-001) [reply src/modules/architect/signup/signup-stepper.tsx:78]: done

File: src/shared/components/controlled-phone-number-input.tsx (hunk 6 above)
Comment (guardian-001) [reply src/shared/components/controlled-phone-number-input.tsx:59]: fixed :
tablet : 
image
phone : 
image

File: src/modules/login/login-form.tsx (hunk 5 above)
Comment (guardian-001) [reply src/modules/login/login-form.tsx:31]: done

File: src/validations/shared/shared-schema.tsx (hunk 7 above)
Comment (guardian-001) [reply src/validations/shared/shared-schema.tsx:46]: I will use it in the future

File: src/modules/architect/signup/signup-stepper.tsx (hunk 3 above)
Comment (guardian-001) [reply src/modules/architect/signup/signup-stepper.tsx:18]: done 
image

**Pull request merged**